### PR TITLE
`NUM_TABLES` constant

### PR DESCRIPTION
**Pull request by wborgeaud**
Add a `NUM_TABLES` constant (replacing the current `Table::num_tables()`) and replace all `Vec<T>` indexed by tables with `[T; NUM_TABLES]`.

### CODE REVIEW

File: evm/src/generation/mod.rs [hunk 1]
```diff
@@ -90,9 +90,16 @@ pub(crate) fn generate_traces<F: RichField + Extendable<D>, const D: usize>(
 
     let cpu_trace = trace_rows_to_poly_values(cpu_rows);
     let keccak_trace = all_stark.keccak_stark.generate_trace(keccak_inputs);
+    let keccak_memory_trace = all_stark.keccak_memory_stark.generate_trace(vec![], 0); // TODO
```
Comment (wborgeaud) [inline evm/src/generation/mod.rs:93]: Not sure which parameters to use here. Do we already have this data here?

Comment (dlubarov) [reply evm/src/generation/mod.rs:93]: Sorry I forgot this, it should pass `operations = keccak_memory_inputs` (from `state` which is unpacked a few lines above), and `min_rows = 1 << cap_height` (looks like we don't have access to that config yet so we'd need to add it). I can add it afterward if it's easier.

**Review by dlubarov — APPROVED** [review]
Good idea, looks like it already caught one issue!

File: evm/src/generation/mod.rs (hunk 1 above)
Comment (wborgeaud) [reply evm/src/generation/mod.rs]: Thanks! Made the change

**Pull request merged**